src/IO/ReadBufferFromEncryptedFile.cpp

```diff
@@ -87,6 +101,9 @@ bool ReadBufferFromEncryptedFile::nextImpl()
         bytes_read += in->read(encrypted_buffer.data() + bytes_read, encrypted_buffer.size() - bytes_read);
     }
 
+    if (old_buffer)
+        in->set(old_buffer->begin(), old_buffer->size());
```

Comment:
hm, I assume we lose here value of
https://github.com/ClickHouse/ClickHouse/blob/484bf00c06ff9a6caefb8d67eab9d43332f4c6b1/src/IO/ReadBuffer.h#L248-L251
from https://github.com/ClickHouse/ClickHouse/blob/484bf00c06ff9a6caefb8d67eab9d43332f4c6b1/src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp#L583

Reply:
Let's add your new configuration into https://github.com/ClickHouse/ClickHouse/blob/484bf00c06ff9a6caefb8d67eab9d43332f4c6b1/tests/queries/0_stateless/02313_filesystem_cache_seeks.sh#L11
https://github.com/ClickHouse/ClickHouse/blob/484bf00c06ff9a6caefb8d67eab9d43332f4c6b1/tests/queries/0_stateless/02241_filesystem_cache_on_write_operations.sh#L10

Reply:
It could also be useful to have a test like https://github.com/ClickHouse/ClickHouse/blob/master/tests/queries/1_stateful/00170_s3_cache.sql
but it is not necessary, up to you, if you want, because this test is quite heavy to prepare
(https://github.com/ClickHouse/ClickHouse/blob/484bf00c06ff9a6caefb8d67eab9d43332f4c6b1/docker/test/stateful/run.sh#L105C24-L106)